internal/sops/awskms/keysource.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awskms
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
```

Comment:
There is a newer SDK available: https://github.com/aws/aws-sdk-go-v2

Reply:
yes, but upstream sops [still use v1](https://github.com/mozilla/sops/blob/master/kms/keysource.go#L17), so i thought it might be better to use that as well to avoid any incompatibility risks 😓

Reply:
That's what the `_Compat` tests are for, the same is true for the Azure KMS integration :-)